### PR TITLE
ACS-2336 POST endpoints definitions for archive/archive-restore content

### PR DESCRIPTION
**Pull request by mpichura**
All changes from this PR are also included in PR#138(https://github.com/Alfresco/rest-api-explorer/pull/138)

### CODE REVIEW

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 1]
```diff
@@ -3174,6 +3184,183 @@ paths:
...
+        It also requires at least one specific implementation of underlying functionality in Cloud Connector(s).
+
+        Request to send given content to acrhive.
+      operationId: requestArchiveContent
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/nodeIdParam'
+        - $ref: '#/parameters/contentPropNameParam'
+      responses:
```
Comment (PunchedChimera) [inline src/main/webapp/definitions/alfresco-core.yaml:3300]: Should we also get a 403 for users without write permission?

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 2]
```diff
@@ -3174,6 +3184,183 @@ paths:
...
+            Normal restore priority request body example:
+            ```JSON
+            {
+              "restorePriority": "Normal"
+            }
+            ```
+          required: false
+          schema:
+            $ref: '#/definitions/RestoreArchivedContentRequest'
+      responses:
```
Comment (PunchedChimera) [inline src/main/webapp/definitions/alfresco-core.yaml:3351]: Should we also get a 403 for users without write permission? (I can't remember what permissions are required on this method so apologies if this is wrong :smile: )

**Review by PunchedChimera — COMMENTED** [review]
Looks, good. Couple of questions. Not sure if they were in the design doc too? That may need updating.

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 1 above)
Comment (mpichura) [reply src/main/webapp/definitions/alfresco-core.yaml:3300]: Thanks I added 403 and 409 and some more description to 400 as well.

File: src/main/webapp/definitions/alfresco-core.yaml (hunk 2 above)
Comment (mpichura) [reply src/main/webapp/definitions/alfresco-core.yaml:3375]: Thanks I added 403 and 409 and some more description to 400 as well.

File: src/main/webapp/definitions/alfresco-core.yaml [hunk 3]
```diff
@@ -3174,6 +3184,215 @@ paths:
...
+            Invalid parameter: nodeId is not a valid format, or is not a file
+                                           property name of content is not valid
+            Content's storage state does not allow restore.
+            Invalid restore paramters.
+        '401':
+          description: Authentication failed
+        '403':
+          description: Not authorized
+        '409':
+          description: Content already restored
```
Comment (PunchedChimera) [inline src/main/webapp/definitions/alfresco-core.yaml:3391]: can be add /restore is in progress?
```suggestion
          description: Content already restored/ Restoration is already in progress
```

**Comment by mpichura** [comment]
Closing as this PR is obsolete now.

**Pull request closed without merging**